oozie/src/main/java/org/apache/falcon/oozie/ImportExportCommon.java

```diff
@@ -70,4 +83,37 @@ static StringBuilder buildUserPasswordArg(StringBuilder builder, StringBuilder s
         }
         return builder;
     }
+
+    public static void addHCatalogProperties(Properties props, Feed entity, Cluster cluster,
+        Path buildPath, WORKFLOWAPP workflow, OozieOrchestrationWorkflowBuilder<Feed> wBuilder)
+        throws FalconException {
+        if (FeedHelper.getStorageType(entity, cluster) == Storage.TYPE.TABLE) {
+            addHCatalogShareLibs(props);
+            addMetastoreURI(props, cluster);
+            if (SecurityUtil.isSecurityEnabled()) {
+                // add hcatalog credentials for secure mode and add a reference to each action
+                wBuilder.addHCatalogCredentials(workflow, cluster,
```

Comment:
Minor nit : This is not a problem in your code. addHCatalogCredentials method uses HiveUtil.java which has a public var METASTROE_URI. If you are submitting another patch, can you  please correct this variable. If not, we can fix it later.